step/step.go

```diff
@@ -42,40 +51,68 @@ func New(
 	pathProvider pathutil.PathProvider,
 	pathModifier pathutil.PathModifier,
 	envRepo env.Repository,
+	derivedDataPathProvider xcodecache.SwiftPackageCache,
 ) SaveCacheStep {
 	return SaveCacheStep{
-		logger:       logger,
-		inputParser:  inputParser,
-		pathChecker:  pathChecker,
-		pathProvider: pathProvider,
-		pathModifier: pathModifier,
-		envRepo:      envRepo,
+		logger:                  logger,
+		inputParser:             inputParser,
+		pathChecker:             pathChecker,
+		pathProvider:            pathProvider,
+		pathModifier:            pathModifier,
+		envRepo:                 envRepo,
+		derivedDataPathProvider: derivedDataPathProvider,
 	}
 }
 
-func (step SaveCacheStep) Run() error {
+func (step SaveCacheStep) ProcessConfig() (Config, error) {
 	var input Input
 	if err := step.inputParser.Parse(&input); err != nil {
-		return fmt.Errorf("failed to parse inputs: %w", err)
+		return Config{}, err
 	}
 	stepconf.Print(input)
+	step.logger.EnableDebugLog(input.Verbose)
+
+	input.DerivedDataPath = strings.TrimSpace(input.DerivedDataPath)
+	input.ProjectPath = strings.TrimSpace(input.ProjectPath)
+	if input.DerivedDataPath == "" && input.ProjectPath == "" {
+		return Config{}, fmt.Errorf("provide either Derived Data Path (derived_data_path) or Xcode Project Path (project_path) Inputs")
+	}
+	if input.DerivedDataPath != "" && input.ProjectPath != "" {
+		step.logger.Warnf("Both Derived Data Path (derived_data_path) and Xcode Project Path (project_path) Inputs are provided, only derived_data_path is used, project_path is ignored")
```

Comment:
Do we really ignore `input.ProjectPath` in this case? In the line 85 it seems we use the project path when provided.

Reply:
Good catch, fixed.